azure-pipelines.yml

```diff
@@ -3,19 +3,112 @@
-  vmImage: "ubuntu-latest"
-
-steps:
-- task: NodeTool@0
-  inputs:
-    versionSpec: "12.x"
-  displayName: Use Node.js 12.x
-- script: npm ci
-  displayName: Install dependencies
-- script: npm audit
-  displayName: Audit dependencies
-- script: npm test
-  displayName: Run unit tests
+stages:
+  - stage: unit_tests
+    displayName: Unit tests
+    jobs:
+      - job: unit_tests
+        displayName: Unit tests
+        strategy:
+          matrix:
+            Linux:
+              imageName: "ubuntu-latest"
+            macOS:
+              imageName: "macOS-latest"
+            Windows:
+              imageName: "windows-latest"
+        pool:
+          vmImage: $(imageName)
+        steps:
+          - task: NodeTool@0
+            inputs:
+              versionSpec: "10.x"
+            displayName: Use Node.js 10.x
+          - script: npm ci
+            displayName: Install dependencies
+          - script: npm audit
+            displayName: Audit dependencies
+          - script: npm test
+            displayName: Run unit tests
+  - stage: integration_tests
+    displayName: Integration tests
+    dependsOn: unit_tests
+    jobs:
+      - job: chaincode_integration_tests
+        displayName: Chaincode integration tests
+        strategy:
+          matrix:
+            Linux:
+              imageName: "ubuntu-latest"
+        pool:
+          vmImage: $(imageName)
+        steps:
+          - task: NodeTool@0
+            inputs:
+              versionSpec: "10.x"
+            displayName: Use Node.js 10.x
+          - script: npm ci
+            displayName: Install dependencies
+          - script: scripts/run-integration-tests.sh chaincode_tests
+            displayName: Run chaincode integration tests
+      - job: contract_integration_tests
+        displayName: Contract integration tests
+        strategy:
+          matrix:
+            Linux:
+              imageName: "ubuntu-latest"
+        pool:
+          vmImage: $(imageName)
+        steps:
+          - task: NodeTool@0
+            inputs:
+              versionSpec: "10.x"
+            displayName: Use Node.js 10.x
+          - script: npm ci
+            displayName: Install dependencies
+          - script: scripts/run-integration-tests.sh contract_tests
+            displayName: Run contract integration tests
+      - job: network_integration_test
+        displayName: Network integration test
+        strategy:
+          matrix:
+            Linux:
+              imageName: "ubuntu-latest"
+        pool:
+          vmImage: $(imageName)
+        steps:
+          - task: NodeTool@0
+            inputs:
+              versionSpec: "10.x"
+            displayName: Use Node.js 10.x
+          - script: npm ci
+            displayName: Install dependencies
+          - script: scripts/run-integration-tests.sh network_test
+            displayName: Run network integration test
+  - stage: publish
+    displayName: Publish to npm
+    dependsOn: integration_tests
```

Comment:
shouldn't this also depend on the unit tests

Reply:
Integration tests depend on unit tests, so the dependency is implicit.